pkg/statesolver/docker.go

```diff
@@ -124,6 +124,30 @@ func (s DockerSolver) fetchContainerLogs(
 	return outbuf, errbuf, nil
 }
 
+func (s DockerSolver) FileExists(
+	ctx context.Context,
+	filepath string,
+	source *builddef.Context,
+) (bool, error) {
+	if source == nil {
+		return false, nil
+	}
+
+	// @TODO: implement a proper way to test if a file exists instead of reading it
+	_, err := s.ReadFile(ctx, filepath, s.FromContext(source))
```

Comment:
why to use the `ReadFile` method from the `DockerSolver` instead of using the `fs` package ?

is it because of the build context (local/git) ?

Reply:
Indeed, that's because the source can be either a local or a git-based context :wink: